File: crossecore-java/src/java_/ManifestAttributeBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ManifestAttributeBase 
extends org.eclipse.emf.ecore.impl.MinimalEObjectImpl.Container implements ManifestAttribute
{
	protected static final java.lang.String KEY_EDEFAULT = "";
	private java.lang.String _key = KEY_EDEFAULT;
	public java.lang.String getKey()
	{
		return _key;
	}
	public void setKey(java.lang.String value){
		
		java.lang.String oldValue = _key;
		_key = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.MANIFESTATTRIBUTE_KEY, oldValue, value));
		
	}
	
	
	
	protected static final java.lang.String VALUE_EDEFAULT = "";
	private java.lang.String _value = VALUE_EDEFAULT;
	public java.lang.String getValue()
	{
		return _value;
	}
	public void setValue(java.lang.String value){
		
		java.lang.String oldValue = _value;
		_value = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.MANIFESTATTRIBUTE_VALUE, oldValue, value));
		
	}
	
	
	

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.MANIFESTATTRIBUTE;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.MANIFESTATTRIBUTE_KEY:
				return getKey();
			case Java_PackageImpl.MANIFESTATTRIBUTE_VALUE:
				return getValue();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.MANIFESTATTRIBUTE_KEY:
				setKey((java.lang.String) newValue);
				return;
			case Java_PackageImpl.MANIFESTATTRIBUTE_VALUE:
				setValue((java.lang.String) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.MANIFESTATTRIBUTE_KEY:
				return KEY_EDEFAULT == null ? _key != null : !KEY_EDEFAULT.equals(_key);
			case Java_PackageImpl.MANIFESTATTRIBUTE_VALUE:
				return VALUE_EDEFAULT == null ? _value != null : !VALUE_EDEFAULT.equals(_value);
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ManifestAttributeBase.ts
import {EClass} from "ecore/EClass";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {Bag} from "ecore/Bag";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ManifestAttribute} from "./ManifestAttribute";
		
			export class ManifestAttributeBase
			extends BasicEObjectImpl
			implements ManifestAttribute
			{
				private _key:string = '';
				get key():string{
					return this._key;
				}
				set key(value:string){
					this._key = value; 
				}
				private _value:string = '';
				get value():string{
					return this._value;
				}
				set value(value:string){
					this._value = value; 
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ManifestAttributeBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.MANIFESTATTRIBUTE_KEY:
							return this.key;
						case Java_PackageLiterals.MANIFESTATTRIBUTE_VALUE:
							return this.value;
					}
					//return this.eGetFromBasicEObjectImpl(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.MANIFESTATTRIBUTE_KEY:
							this.key = <string> newValue;
							return;
						case Java_PackageLiterals.MANIFESTATTRIBUTE_VALUE:
							this.value = <string> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
